### PR TITLE
added metadata to twin_axis.py

### PR DESCRIPTION
**Pull request by PatChizzy**
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [ ] issues: fixes #11765
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: examples/models/transform_jitter.py [hunk 1]
```diff
@@ -1,3 +1,18 @@
+"""
+==========================================
+Jittered Scatter Plot with Bokeh and NumPy
+==========================================
```
Comment (tcmetzger) [inline examples/models/transform_jitter.py]: You don't need to add a headline here!

```suggestion
```

File: examples/models/transform_jitter.py [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+
+This example demonstrates how to add jitter to a scatter plot using Bokeh and NumPy.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.circle, bokeh.models.sources.ColumnDataSource, bokeh.models.markers.Circle, bokeh.models.Jitter, bokeh.models.LabelSet, bokeh.models.widgets.Button, bokeh.models.callbacks.CustomJS
+    :refs: :ref:`userguide_plotting` > :ref:`userguide_plotting` > :ref:`userguide_widgets`, :ref:`userguide_styling` > :ref:`userguide_annotations`
```
Comment (tcmetzger) [inline examples/models/transform_jitter.py]: These lines are too long (this is also why the CI codebase tests are failing, https://github.com/bokeh/bokeh/actions/runs/4404874372/jobs/7715229116), please add some line breaks to stay below the max of 165 characters per line!

Comment (tcmetzger) [inline examples/models/transform_jitter.py]: These references need to match the exact references defined in the RST files for the docs. For example, in basic.rst, the label is ``ug_basic``, so this is what should go here (see the example in https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations).
These references here are also supposed to work like breadcrumbs, i.e. first the general section (like `ug_interaction`) and then the specific subsection (like `ug_interaction_widgets`

```suggestion
    :refs: :ref:`ug_interaction` > :ref:`ug_interaction_widgets`, :ref:`ug_topics` > :ref:`ug_topics_categorical_scatters_jitter`
```

File: examples/models/transform_jitter.py [hunk 3]
```diff
@@ -1,3 +1,18 @@
+"""
+==========================================
+Jittered Scatter Plot with Bokeh and NumPy
+==========================================
+
+This example demonstrates how to add jitter to a scatter plot using Bokeh and NumPy.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.circle, bokeh.models.sources.ColumnDataSource, bokeh.models.markers.Circle, bokeh.models.Jitter, bokeh.models.LabelSet, bokeh.models.widgets.Button, bokeh.models.callbacks.CustomJS
```
Comment (tcmetzger) [inline examples/models/transform_jitter.py]: I think this path needs to be updated:

```suggestion
    :apis: bokeh.plotting.figure.circle, bokeh.models.sources.ColumnDataSource, bokeh.models.Circle, bokeh.models.Jitter, bokeh.models.LabelSet, bokeh.models.Button, bokeh.models.CustomJS
```

Use the reference guide at https://docs.bokeh.org/en/latest/docs/reference.html to check the paths to use here (usually the parts after the `#` in the URL), e.g. `bokeh.models.RadioGroup` from `https://docs.bokeh.org/en/latest/docs/reference/models/widgets/groups.html#bokeh.models.RadioGroup`

**Review by tcmetzger — CHANGES_REQUESTED** [review]
Thank you!
I added some comments to the first file - most of that also applies to the second file!

**Comment by PatChizzy** [comment]
@tcmetzger thank you for your review. it was super helpful..

Could you please help review this new commit for transform_jitter.py

its failing for docker wheel and mac os. im not sure how to resolve it..

Thank you

**Review by tcmetzger — APPROVED** [review]
This looks great, very helpful!

**Comment by tcmetzger** [comment]
For docs: Yes, I think that https://docs.bokeh.org/en/latest/docs/dev_guide/setup.html#troubleshooting could work as the URL, and we can add this information at the top. We should probably add some subheadings or a bulleted list to organize the different troubleshooting items in that section. And adding a screenshot would definitely be helpful!

**Review by tcmetzger — CHANGES_REQUESTED** [review]
Sorry, mixed up my PRs here. The jitter file looks great - let's try to also update the twin_axis file in a similar way, that'll be great!

I believe the tests failing are unrelated!

**Comment by PatChizzy** [comment]
> Sorry, mixed up my PRs here. The jitter file looks great - let's try to also update the twin_axis file in a similar way, that'll be great!
> 
> I believe the tests failing are unrelated!

@tcmetzger twin_axis is updated.

I would love to work on more of these. 
Could you please give me pointers on how to prevent the failures for mac os and docker wheel?

**Comment by PatChizzy** [comment]
> 

I changed from double quotes to single quotes to be conventionally appealing

File: examples/models/transform_jitter.py [hunk 4]
```diff
@@ -1,3 +1,12 @@
+''' This example demonstrates how to add jitter to a scatter plot using Bokeh and NumPy.
```
Comment (bryevdv) [inline examples/models/transform_jitter.py]: Numpy is not really a focus of this example, it's just used to make the raw coords, not the jitter. Suggest this change:
```suggestion
''' This example demonstrates how to us a jitter transform on coordinate data.

File: examples/models/twin_axis.py [hunk 5]
```diff
@@ -1,3 +1,12 @@
+''' This example demonstrates how to create a plot with two y-axes using Bokeh and NumPy.
```
Comment (bryevdv) [inline examples/models/twin_axis.py]: "with Bokeh" is implicit, and numpy usage is really just incidental. Suggest:
```suggestion
''' This example demonstrates how to create a plot with two y-axes.
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
Just a few small comments!

File: examples/models/twin_axis.py (hunk 5 above)
Comment (PatChizzy) [reply examples/models/twin_axis.py]: Oh, thank you @bryevdv 

I really appreciate the comments and feedback, its a learning curve for me

File: examples/models/transform_jitter.py [hunk 6]
```diff
@@ -1,3 +1,12 @@
+''' This example demonstrates how to us a jitter transform on coordinate data.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.circle, bokeh.models.sources.ColumnDataSource, bokeh.models.Circle, bokeh.models.Jitter
+    :refs: :ref:`ug_interaction` > :ref:`ug_interaction_widgets` > :ref:`ug_topics` > :ref:`ug_topics_categorical_scatters_jitter`
```
Comment (bryevdv) [inline examples/models/transform_jitter.py]: Ah sorry I missed this earlier, we stopped trying to do the "breadcrumbs" chain of links, it's just too hard to maintain by hand. All we need are the terminal section references (seperated by commas)
```suggestion
    :refs: :ref:`ug_interaction_widgets`, :ref:`ug_topics_categorical_scatters_jitter`
```

File: examples/models/twin_axis.py [hunk 7]
```diff
@@ -1,3 +1,12 @@
+''' This example demonstrates how to create a plot with two y-axes.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.circle, bokeh.models.sources.ColumnDataSource, bokeh.models.markers.Circle
+    :refs: :ref:`userguide_plotting` > :ref:`userguide_plotting_markers` > :ref:`userguide_plotting` > :ref:`userguide_plotting_extra_axes`
```
Comment (bryevdv) [inline examples/models/twin_axis.py]: same comment
```suggestion
    :refs: :ref:`userguide_plotting_markers`, :ref:`userguide_plotting_extra_axes`
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
Thanks @PatChizzy one final comments I  missed earlier, we started using a simpler format for the references. After this change it will be good to merge!

**Review by bryevdv — APPROVED** [review]

**Comment by PatChizzy** [comment]
@bryevdv @tcmetzger i changed the sunrise color from orange to dark orange, to distinguish it from the yellow patch representing daylight.. 
In line with microtask 7
image

**Comment by bryevdv** [comment]
@PatChizzy I think the metadata changes look good, can you revert the color change to the daylight example? Generally we try to keep PRs focused to single topics. 

Also while I think that color change *is* an improvement, I think if we checked a contrast checker we'd find that it is still not great.

**Comment by PatChizzy** [comment]
> @PatChizzy I think the metadata changes look good, can you revert the color change to the daylight example? Generally we try to keep PRs focused to single topics.
> 
> Also while I think that color change _is_ an improvement, I think if we checked a contrast checker we'd find that it is still not great.

Hi @bryevdv this is well understood. Change is reverted.

Should I close this PR or leave it open?

**Review by tcmetzger — APPROVED** [review]

**Comment by tcmetzger** [comment]
@PatChizzy Thank you, I'll merge (and, thus, close) this PR with the metadata update!

It would be great if you could open a new PR focused on the sunrise color!

**Pull request merged**